tests/Endpoints/SessionObjectiveTest.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Tests\Endpoints;
+
+use App\Tests\DataLoader\ObjectiveData;
+use App\Tests\DataLoader\SessionData;
+use App\Tests\DataLoader\TermData;
+use App\Tests\ReadWriteEndpointTest;
+
+/**
+ * SessionObjectiveTest API endpoint Test.
+ * @group api_1
+ */
+class SessionObjectiveTest extends ReadWriteEndpointTest
+{
+    protected $testName =  'sessionObjectives';
+
+    /**
+     * @inheritdoc
+     */
+    protected function getFixtures()
+    {
+        return [
+            'App\Tests\Fixture\LoadObjectiveData',
+            'App\Tests\Fixture\LoadTermData',
+            'App\Tests\Fixture\LoadCourseData',
+            'App\Tests\Fixture\LoadSessionData',
+            'App\Tests\Fixture\LoadProgramYearData',
+            'App\Tests\Fixture\LoadSessionObjectiveData',
+            'App\Tests\Fixture\LoadCourseObjectiveData',
+            'App\Tests\Fixture\LoadProgramYearObjectiveData',
+        ];
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function putsToTest()
+    {
+        return [
+            'session' => ['session', 2],
+            'objective' => ['objective', 2],
+            'terms' => ['terms', [1, 4]],
+        ];
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function readOnlyPropertiesToTest()
+    {
+        return [];
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function filtersToTest()
+    {
+        return [
+            'id' => [[0], ['id' => 1]],
+            'ids' => [[1, 2], ['id' => [2, 3]]],
+            'session' => [[1, 2], ['session' => 4]],
+            'objective' => [[2], ['objective' => 7]],
```

Comment:
We'll need to be able to filter these by course it in order to pre-load on the frontend.

Reply:
i've added that filter.